### DISCUSSION TITLE
OpenDTU-OnBattery + Pylontech Force H2

### DISCUSSION THREAD

**Discussion by jkgw2022** [hoylabs/OpenDTU-OnBattery#1447] [Q&A]
Hallo zusammen,

ich habe einen Pylontech Force H2 und einen Hoymiles HAT-AV-EUG1, die über Can-Bus miteinander verbunden sind.

An die Can-Bus-Verbindung habe ich einen OpenDTU Fusion mit CAN/Iso Shield angeschlossen, wobei auf dem OpenDTU Fusion das aktuellste OpenDTU-OnBattery installiert ist. Pin-Mapping wurde gemäß "https://github.com/markusdd/OpenDTUFusionDocs/blob/main/CANIso.md" geändert.

Danach wurde unter den "Batterie Einstellungen" bei Daten-Anbieter "Pylontech per Can-Bus" ausgewählt.

Das Ergebnis ist, dass die Live-Ansicht nichts anzeigt.

Die Konsole liefert sowas:

09:13:59.626 > [Pylontech] Received CAN message: 0x4200 - 00 00 00 00 00 00 00 00
09:13:59.753 > [Pylontech] Received CAN message: 0x4211 - 82 07 30 75 14 05 0A 64
09:13:59.957 > [Pylontech] Received CAN message: 0x4221 - 52 08 CC 06 E9 75 77 74
09:14:00.162 > [Pylontech] Received CAN message: 0x4231 - 8E 0C 67 0C 1E 00 06 00
09:14:00.368 > [Pylontech] Received CAN message: 0x4241 - C4 04 BA 04 25 00 00 00
09:14:00.572 > [Pylontech] Received CAN message: 0x4261 - FF FF FF FF 01 00 00 00
09:14:01.951 > [Pylontech] Received CAN message: 0x4200 - 02 00 00 00 00 00 00 00
09:14:02.001 > [Pylontech] Received CAN message: 0x7311 - 02 00 1E 03 01 06 35 0C
09:14:02.211 > [Pylontech] Received CAN message: 0x7321 - 3C 00 02 1E C0 00 25 00
09:14:02.269 > [Pylontech] Received CAN message: 0x7331 - 50 59 4C 4F 4E 54 45 43
09:14:02.417 > [Pylontech] Received CAN message: 0x7341 - 48 00 00 00 00 00 00 00
09:14:04.276 > [Pylontech] Received CAN message: 0x4200 - 00 00 00 00 00 00 00 00
09:14:04.338 > [Pylontech] Received CAN message: 0x4211 - 82 07 30 75 14 05 0A 64
09:14:04.464 > [Pylontech] Received CAN message: 0x4221 - 52 08 CC 06 E9 75 77 74
09:14:04.670 > [Pylontech] Received CAN message: 0x4231 - 8E 0C 67 0C 1E 00 06 00
09:14:04.874 > [Pylontech] Received CAN message: 0x4241 - C4 04 BA 04 25 00 00 00
09:14:05.079 > [Pylontech] Received CAN message: 0x4251 - 13 02 00 00 00 00 00 00
09:14:06.605 > [Pylontech] Received CAN message: 0x4200 - 02 00 00 00 00 00 00 00
09:14:06.716 > [Pylontech] Received CAN message: 0x7311 - 02 00 1E 03 01 06 35 0C
09:14:06.921 > [Pylontech] Received CAN message: 0x7321 - 3C 00 02 1E C0 00 25 00
09:14:06.990 > [Pylontech] Received CAN message: 0x7331 - 50 59 4C 4F 4E 54 45 43
09:14:07.126 > [Pylontech] Received CAN message: 0x7341 - 48 00 00 00 00 00 00 00

Und dann gehts wieder von vorne los.

OpenDTU-OnBattery scheint ja in erster Linie für Niederspannungs-Akkus gemacht.
Ich hatte gehofft, meinen Hochvolt-Akku auslesen zu können.

Kann hier jemand helfen?

Gruß
Jürgen

**Comment by AndreasBoehm** [comment]
Hallo Jürgen, 
gibt es eine CAN Protokoll Dokumentation für den Force H2? 
Im Log ist zu sehen das Daten ankommen aber wir diese nicht verarbeiten da wir die Adressen nicht zuordnen können, somit siehst du auch in der Live View nichts.

Grüße,
Andreas

**Reply by jkgw2022** [reply]
Hallo Andreas,

danke für Deine Antwort.
Laut meinen Recherchen sollte die beigefügte Datei helfen können.

Hilft diese weiter?
[CAN-Bus-Protocol-Sermatec-high-voltage-V1.1810kW.pdf](https://github.com/user-attachments/files/18135045/CAN-Bus-Protocol-Sermatec-high-voltage-V1.1810kW.pdf)

Grüße
Jürgen

**Reply by AndreasBoehm** [reply]
Das sieht gut aus! Ich hab aber nur kurz die Adressen abgeglichen, nicht den Inhalt der Nachrichten angeschaut.

Hab jetzt erst realisiert das es sich um High Voltage Akkus handelt. Die passen ja nicht so wirklich zu den anderen Produkten die wir in diesem Projekt unterstützen.

Kannst du mir mehr über deinen Anwendungsfall erzählen?

**Reply by jkgw2022** [reply]
Nun, ich lese per OpenDTU-OnBattery meine diversen Hoymiles Wechselrichter (HM-800, HM-1500, HMT-2250) aus. Per MQTT speise ich deren Daten in Home Assistant und da lassen sich dann viele schöne Dinge machen, wie z. B. eine simple Langzeitüberwachung mit Grafana oder Schaltaufgaben ;-)

Im Endeffekt würde ich nun gerne Daten meines Hochvolt-Akku - Pylontech Force H2 - von OpenDTU-onBattery über MQTT in Home Assistant spielen.

Wahnsinn wäre, wenn ich auch noch den Hoymiles HAT-AV-EUG1 auslesen könnte und dessen Daten per MQTT in Home Assistant spielen könnte.

**Reply by AndreasBoehm** [reply]
Verstehe, da High Voltage Akkus nicht wirklich zu unserem Anwendungsfall passen muss ich dich bitten die Implementierung des Force H2 Supports selbst durch zu führen. Vielleicht wirst du alternativ bei ESPHome fündig.

**Reply by jkgw2022** [reply]
Wenn ich das in OpenDTU-OnBattery verwirklichen will, wo kann ich da am besten ansetzen? Hast Du einen Tipp für mich?

**Reply by jkgw2022** [reply]
Auch verstehe ich nicht ganz, warum das hier nicht der Anwendungsfall ist.
Der einzige Unterschied zu den Anwendungsfällen (oder besser zum letzten Anwendungsfall auf dieser Seite: https://github.com/hoylabs/OpenDTU-OnBattery/wiki) hier ist Hochvolt anstelle von Niedervolt; und der andere Wechselrichter.
Aber wenn mein Akku voll ist, soll die Einspeisung runtergeregelt werden.

**Reply by AndreasBoehm** [reply]
Solange der Wechselrichter nicht unterstützt wird ist die integration eines HV Akkus ziemlich sinnlos für das Projekt.

Wenn du dich einbringen möchtest solltest du mit dem Wechselrichter anfangen.

Du kannst natürlich auch deinen eigenen Fork des Projekts anlegen und dort die HV Akku integration pflegen.

Hier findest du die implementierung der Pylontech LV Akkus: https://github.com/hoylabs/OpenDTU-OnBattery/blob/development/src/PylontechCanReceiver.cpp
